### PR TITLE
QEP #14 - Authentication configuration system with master password

### PR DESCRIPTION
**Pull request by dakcarto**
Direct [link to QEP .rst preview](https://github.com/dakcarto/QGIS-Enhancement-Proposals/blob/auth-system/qep-14-authentication-system.rst).

### CODE REVIEW

**Comment by NathanW2** [comment]
Gave it #14

**Comment by NathanW2** [comment]
@dakcarto has there been any more work done on this?  Are you using it in production.

**Comment by dakcarto** [comment]
Please note, this QEP is currently being revamped to address the large number of changes in the #2330 PR, specifically the new authentication editors and method plugins. It should be done today.

**Comment by dakcarto** [comment]
@NathanW2 This is a final draft (unless you see errors), ready to vote on. A bit of a moot point since I already completed most of the work and it has been heavily tested for months. I guess the vote is more of whether the approach is sound, which I think it is.

Btw, the authentication system now supports authentication method plugins (kind of like data provider).

File: qep-14-authentication-system.rst [hunk 1]
```diff
@@ -0,0 +1,503 @@
...
+  confirm before operation)
+* Add ability to change, edit or remove authcfg from existing layer in
+  properties dialog
+* Add simple read-only text field in layer properties dialog to quickly copy
+  authcfg
+* Add copy/paste/add/edit/remove of authcfg in layer contextual menu in Legend
+  panel
+* On layer load, notify user if any associated authcfg is missing in auth
+  database
+* Add layer authcfg (re)assignment in Handle Bad Layers dialog (can currently
```
Comment (nyalldawson) [inline qep-14-authentication-system.rst]: Good to see this listed - I think these two tasks are important and should be prioritised  if possible.

**Comment by dakcarto** [comment]
Hmm. A little dismayed this has been up since January with almost no action or comments about the design or implementation from the PSC, and again with the major revamp within the past week.

Can I get some opinions here when you get a chance? Is it good enough I can just push the fully functioning PR and forego the whole QEP process?

**Comment by NathanW2** [comment]
Larry feel free to just focus on the PR and merge if you have good feedback.

I need to revamp the QEP process for faster processing.

**Comment by NathanW2** [comment]
Leave this here though because we can use it for docs later.

**Comment by dakcarto** [comment]
Ok. Thanks @NathanW2 for getting back to me.

**Comment by NathanW2** [comment]
Sorry for delay and stuffing around

**Comment by timlinux** [comment]
Hi

> On 25 Sep 2015, at 02:44, Larry Shaffer <notifications@github.com> wrote:
> 
> Hmm. A little dismayed this has been up since January with almost no action or comments about the design or implementation from the PSC, and again with the major revamp within the past week.
> 
> Can I get some opinions here when you get a chance? Is it good enough I can just push the fully functioning PR and forego the whole QEP process?
> 
> 

In general you should solicit comments from the community of developers and consider the PSC as a source of arbitration if there are disputes over decisions being made - or if you feel your QEP is not getting the attention it needs ask us to help - though I think Nathan is on it now. Nathan would normally bring any QEP needing a vote from the PSC to us at that stage.

Apologies for any confusion!

Regards

Tim
> —
> Reply to this email directly or view it on GitHub <https://github.com/qgis/QGIS-Enhancement-Proposals/pull/17#issuecomment-143030130>.
> 

—





Tim Sutton

Visit http://kartoza.com <http://kartoza.com/> to find out about open source:

* Desktop GIS programming services
* Geospatial web development
* GIS Training
* Consulting Services

Skype: timlinux Irc: timlinux on #qgis at freenode.net
Tim is a member of the QGIS Project Steering Committee

Kartoza is a merger between Linfiniti and Afrispatial

**Comment by dakcarto** [comment]
Hi Tim,

Thank you for the clarification.

**Comment by NathanW2** [comment]
QEPs done via issues now.

New ticket at #35 

Sorry for the inconvenience.

Email out soon on why.

**Pull request closed without merging**